worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -30,28 +60,136 @@ public Job putJobInProgress(String jobId) {
         final Job job = jobRepository.findByJobUuid(jobId);
         Assert.notNull(job, String.format("Job %s not found", jobId));
 
-
         // validate status is SUBMITTED
         if (!SUBMITTED.equals(job.getStatus())) {
-            final String errMsg = String.format("Job %s is not in %s status. Skipping job", jobId, SUBMITTED);
+            final String errMsg = String.format("Job %s is not in %s status.", jobId, SUBMITTED);
             throw new IllegalArgumentException(errMsg);
         }
 
         job.setStatus(IN_PROGRESS);
 
-        log.info("Job [{}] is now IN_PROGRESS", job.getId());
         return jobRepository.save(job);
     }
 
     @Override
-    public void completeJob(Job job) {
+    public Job processJob(final String jobId) {
+
+        final Job job = jobRepository.findByJobUuid(jobId);
+        log.info("Found job : {} - {} - {} ", job.getId(), job.getJobUuid(), job.getStatus());
+
+        final Sponsor sponsor = job.getUser().getSponsor();
+
+        final List<Contract> attestedContracts = sponsor.getAttestedContracts();
+        log.info("number of attested contracts : {}", attestedContracts.size());
+
+
+        final Path outputDirPath = Paths.get(efsMount, job.getJobUuid());
+        final Path outputDir = fileService.createDirectory(outputDirPath);
+        for (Contract contract : attestedContracts) {
+            log.info("contract : {} ", contract.getContractNumber());
+
+            JobOutput jobOutput = null;
+            try {
+                jobOutput = processContract(outputDir, contract);
+                jobOutput.setError(false);
+            } catch (Exception e) {
+                jobOutput.setError(true);
+                log.error("error processing contract : {} ", contract.getContractNumber(), e);
+            }
+
+            job.addJobOutput(jobOutput);
+            jobOutputRepository.save(jobOutput);
+        }
```

Comment:
My understanding is that each contract should have a row in the `job_output` table. Is that correct?

Reply:
@spathiyil For the MVP, a job might have up to **2** outputs. Consider this: you are processing a contract with 100 patients, out of which 1 failed (a non-fatal condition). You would still write a _success_ `JobOutput` with `ndjson` with EOB resources for the 99 patients, but you'd also write an _error_ `JobOutput` with `ndjson` with a single `OperationOutcome` entry that explains what happened to the failed patient.

Reply:
done